CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+# Contribute to WarpX
+
+We welcome new contributors! Here is how to participate to the WarpX 
+development.
+
+## Git workflow
+
+The WarpX project uses [git](https://git-scm.com) for version control. If you 
+are new to git, you can follow one of these tutorials:
+- [Learn git with bitbucket](https://www.atlassian.com/git/tutorials/learn-git-with-bitbucket-cloud)
+- [git - the simple guide](http://rogerdudler.github.io/git-guide/)
+
+### Create your branch from the `dev` branch
+
+Follow instructions from the 
+[WarpX documentation](https://ecp-warpx.github.io/doc_versions/dev/building/building.html) 
+to install the code. Make sure you are on WarpX `dev` branch with
+```
+git checkout dev
+```
+in the WarpX directory.
+
+Create a branch `<branch_name>` (the branch name should reflect the piece 
+of code you want to add, like `fix_spectral_solver`) with
+```
+git checkout -b <branch_name>
+```
+and do the coding you want. It is probably a good time to look at the 
+[AMReX documentation](https://amrex-codes.github.io/amrex/docs_html/). 
+Add the files you work on to the git staging area with 
+```
+git add file_I_created and_file_I_modified
+```
+### Commit & push your changes
+
+Periodically commit your changes with
+```
+git commit -m "This is a 50-char description to explain my work"
+```
+
+The commit message (between quotation marks) is super important to follow the 
+developments and identify bugs.
+
+For the moment, commits are on your local repo only. You can push them to 
+the WarpX Github repo with
+```
+git push -u origin <branch_name>
+```
+
+If you want to synchronize your branch with the `dev` branch (this is useful 
+when the `dev` branch is being modified while you are working on 
+`<branch_name>`), you can use
+```
+git pull --rebase origin dev
+```
```

Comment:
Although I am a very big fan of `rebase`, I am wondering whether this might confuse potential contributors that have a limited experience with `git`. (e.g. because `rebase` goes through multiple commits and forces the developer to fix conflict for **each** of these commits separately).

One alternative would be simply `git pull origin dev` (i.e. a simple merge). @MaxThevenet What do you think?

Reply:
Also: should we add 
```
and fix any conflict that may occur.
```
to the above text?

Reply:
OK I agree, let us keep it simpler